### PR TITLE
SRD redshift distributions

### PR DESCRIPTION
**Pull request by nikosarcevic**
Hi, @elisachisari and @chrgeorgiou ,

As promised, I've created a branch where I added the code to generate the SRD redshift distributions (for lenses and sources for Y1 and Y10).

There is a parameter file (in parameters folder), py file with the code and an example notebook.
Let me know if you want more functionality added or you are ok with how it is.

Niko

### CODE REVIEW

**Comment by nikosarcevic** [comment]
> Hi, @elisachisari and @chrgeorgiou ,
> 
> As promised, I've created a branch where I added the code to generate the SRD redshift distributions (for lenses and sources for Y1 and Y10).
> 
> There is a parameter file (in parameters folder), py file with the code and an example notebook. Let me know if you want more functionality added or you are ok with how it is.
> 
> Niko

I have now added the binning procedure to this branch, as per @heatherprince request.

File: parameters/lsst_desc_parameters.yaml
Comment (elisachisari) [inline parameters/lsst_desc_parameters.yaml:1]: There is a lot here we don't seem to need for the redshift distributions. What was the motivation for including all the parameters?

File: Generate the SRD redshift distributions.ipynb
Comment (elisachisari) [inline Generate the SRD redshift distributions.ipynb:1]: Could you explain somewhere how these were validated? Was there any significant difference with the SRD? To what tolerance do they match?

**Review by elisachisari — CHANGES_REQUESTED** [review]
Runs for me without issues. Some comments. 
Could you also add somewhere in the notebooks what the output files are (.csv) so people notice they are there and also what is contained there?

File: parameters/lsst_desc_parameters.yaml
Comment (nikosarcevic) [reply parameters/lsst_desc_parameters.yaml:1]: motivation is to have the SRD baseline params somewhere. I do plan to convert it to maybe a pdf or something else? I can remove the unnecessary params, not a problem

File: Generate the SRD redshift distributions.ipynb
Comment (nikosarcevic) [reply Generate the SRD redshift distributions.ipynb:1]: Hi, Elisa

I have finally added the validation notebook https://github.com/LSSTDESC/CCLX/blob/SRD-Redshift-Distributions/SRD_binning_validation.ipynb

As explained in there, when I was constructing the pipeline, I have used some code from Danielle Leonard as inspiration (developed by her and Markus Rau). I am comparing the output of the SRD binning from FISK and the output I have from Danielle. She does not have the code online so I am not sure how to tackle this. Maybe we can ask her to confirm.

I have performed some audition tests as you will see in the notebook. 

After you give the blessing we can either keep this code and data somewhere or we can remove it and I can store it on my GH repo.

Thank you for being patient with this!
Niko

File: parameters/lsst_desc_parameters.yaml
Comment (elisachisari) [reply parameters/lsst_desc_parameters.yaml:1]: Yes, I think this belongs in the forecasting repo insted

File: Generate the SRD redshift distributions.ipynb
Comment (elisachisari) [reply Generate the SRD redshift distributions.ipynb:1]: Thanks for this - I see now a validation of the source bins and n(z). What about the validation for the lens bins? Do you think we should add something?

Comment (nikosarcevic) [reply Generate the SRD redshift distributions.ipynb:1]: Hi, Elisa

parameters (yaml) should be here as the code is reading from it.

Re lenses -- I have nothing to compare it to but I have followed the instructions from the SRD (equidistant spacing etc). I do not know if anyone has something I can compare to?

Comment (nikosarcevic) [reply Generate the SRD redshift distributions.ipynb:1]: I would trust it but of course if anyone want to double check the lens bins, I would appreciate that as a sanity check! :)

Comment (nikosarcevic) [reply Generate the SRD redshift distributions.ipynb:1]: I have obtained a code for the SRD distributions from @chrgeorgiou and produced the lens and soruce bins for SRD Y1 and Y10. Then I have compared them to the lens and source bins from this code (adapted FISK code). As you can see in the plots attached, I am getting a 1:1 agreement.
All issues are now resolved, I believe and @elisachisari  can bless this merge.

NOTE: FISK bins are the bins from my code, DL refers to the bins from Danielle Leonard and CG stands for Christos Georgiou.
Screenshot 2023-10-03 at 23 54 38
Screenshot 2023-10-03 at 23 57 46

Comment (elisachisari) [reply Generate the SRD redshift distributions.ipynb:1]: Thanks for checking this too

**Review by elisachisari — APPROVED** [review]
All suggested changes have been incorporated and new validation tests produced as requested

**Pull request merged**